### PR TITLE
Modify datagrid numeric filter input to have units dropdown

### PR DESCRIPTION
**Pull request by ps37**
This will add units dropdown next to number inputs of from and to conditionally. The condition is based on units input.

### CODE REVIEW

File: projects/components/src/form/form-input/form-input.component.spec.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { FormInputComponent, getFormattedDateValue } from './form-input.componen
 export class VcdFormInputWidgetObject extends WidgetObject<FormInputComponent> {
     static tagName = `vcd-form-input`;
 
-    private get inputElement(): HTMLInputElement {
+    get inputElement(): HTMLInputElement {
```
Comment (juanmendes) [inline projects/components/src/form/form-input/form-input.component.spec.ts:15]: Why are we exposing an HTML input element? We typically expose strings, booleans, number

File: projects/components/src/form/form-input/form-input.component.ts [hunk 2]
```diff
@@ -71,6 +81,19 @@ export class FormInputComponent extends BaseFormControl implements AfterViewInit
      */
     @ViewChild('textInput', { static: true }) textInput: ElementRef;
 
+    /**
+     * Emitted when enter key is released for callers to handle the event like... closing a filter widget
+     * {@link DatagridNumericFilterComponent.close}
+     */
+    // tslint:disable-next-line:no-output-on-prefix
```
Comment (juanmendes) [inline projects/components/src/form/form-input/form-input.component.ts:88]: Why break ng's standards?  How about `enterClicked` and `escapeClicked`.

File: projects/components/src/datagrid/filters/datagrid-numeric-filter.component.html [hunk 3]
```diff
@@ -1,15 +1,45 @@
...
+            [inputValueUnit]="unit"
+        >
+        </vcd-number-with-unit-form-input>
+        <vcd-form-input
+            *ngIf="!unit"
+            [type]="'number'"
+            [label]="'vcd.cc.numeric.filter.to' | translate"
+            [formControlName]="'to'"
+            [maxlength]="maxNumberLength"
+            (onEnter)="close()"
```
Comment (juanmendes) [inline projects/components/src/datagrid/filters/datagrid-numeric-filter.component.html:41]: Why do enter and escape do the same thing? Enter should commit the change and escape should revert changes, was this a new feature you just added?

Comment (ps37) [reply projects/components/src/datagrid/filters/datagrid-numeric-filter.component.html:41]: No. The functionality to close the filter widget upon pressing both Enter and Escape keys is there in Vcd UI

File: projects/components/src/datagrid/filters/datagrid-numeric-filter.component.spec.ts [hunk 4]
```diff
@@ -53,24 +98,52 @@ describe('Datagrid numeric filter', () => {
...
+        describe('with units', () => {
+            beforeEach(function(this: HasFinderAndFilter): void {
+                const finderAndFilter = createDatagridFilterTestHelperWithFinder(DatagridNumericFilterComponent, {
+                    unitOptions: [...Bytes.types],
+                    unit: Bytes.MB,
+                });
+                this.finder = finderAndFilter.finder;
+                this.filter = finderAndFilter.filter;
+            });
+            // tslint:disable-next-line:max-line-length
```
Comment (juanmendes) [inline projects/components/src/datagrid/filters/datagrid-numeric-filter.component.spec.ts:131]: Please avoid this `// tslint:disable-next-line:max-line-length`, break up your string if necessary

**Pull request merged**